azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -105,7 +127,14 @@ def __init__(self, incarnation, xml_text):
         super(_ExtensionsGoalStateFromExtensionsConfig, self).__init__()
         self._id = incarnation
         self._text = xml_text
+        self._parse_extensions_config(xml_text)
 
+        try:
+            self._do_common_validations()
+        except Exception as e:
+            raise ExtensionsConfigError("Error parsing ExtensionsConfig (incarnation: {0}): {1}\n{2}".format(incarnation, format_exception(e), xml_text))
```

Comment:
Should we trim the xml text? Incase its very big? Is there a chance `xml_text` can contain secrets? We should redact it in that case too